pkg/endpoint/log.go

```diff
@@ -114,3 +139,61 @@ func (e *Endpoint) UpdateLogger(fields map[string]interface{}) {
 
 	atomic.StorePointer(&e.logger, unsafe.Pointer(l))
 }
+
+func (e *Endpoint) updatePolicyLogger(fields map[string]interface{}) {
```

Comment:
The caller need to hold `Endpoint.mutex` for writing correct? If yes, can we add a comment please?

About `UpdateLogger` comment on holding the mutex, my understanding is that the caller must hold _`Endpoint.mutex`_ (not `Endpoint.Mutex`) for _writing_ (not for reading) even when `fields` is `nil` as we can reach the `atomic.StorePointer` at line 140.

Reply:
I'll update the locking comments ;-)